test/utils/unit-test-helpers.js

```diff
@@ -0,0 +1,168 @@
+ */
+module.exports.checkMediaHeaders = function(createRequestMock, accept, contentType) {
+  const headers = createRequestMock.mock.calls[0][0].defaultOptions.headers;
+  expect(headers.Accept).toEqual(accept);
+  expect(headers['Content-Type']).toEqual(contentType);
+};
+
+/**
+ * Takes the mock object for the `createRequest` method, extracts the headers that were sent with the call,
+ * and checks for the expected value for a user-defined header. This is verify that the SDK accepts header
+ * parameters and sends them as headers in the request.
+ *
+ * @param {Object} createRequestMock - the jest mock object for the `createRequest` method in the `RequestWrapper` class
+ * @param {String} userHeaderName - the name of the header passed by the user, e.g. `Contained-Content-Type`
+ * @param {String} userHeaderValue - the expected value for the header passed by the user
+ * @returns {void}
+ */
+module.exports.checkUserHeader = function(createRequestMock, userHeaderName, userHeaderValue) {
+  const headers = createRequestMock.mock.calls[0][0].defaultOptions.headers;
+  expect(headers[userHeaderName]).toEqual(userHeaderValue);
+};
+
+/**
+ * This method simply ensures that the method executed without any issues by extracting
+ * the argument from the mock object for the `createRequest` method and verifying that it is an object.
+ *
+ * @param {Object} createRequestMock - the jest mock object for the `createRequest` method in the `RequestWrapper` class
+ * @returns {void}
+ */
+module.exports.checkForSuccessfulExecution = function(createRequestMock) {
+  const sdkParams = createRequestMock.mock.calls[0][0];
+  expect(typeof sdkParams).toEqual('object');
+};
+
+/**
+ * Verifies that when a user does not pass in any arguments, a local variable is created as an empty object
+ * and that variable is used in place of the user object in the code. This is accomplished using a mock object
+ * for the `getMissingParams` function, which receives the variable representing user input as an argument.
+ *
+ * @param {Object} missingParamsMock - the jest mock object for the `getMissingParams` function in  the `helpers` module
+ * @returns {void}
+ */
+module.exports.checkUserParamsIsEmptyObject = function(missingParamsMock) {
+  // get arg to getMissingParams
+  const userParams = missingParamsMock.mock.calls[0][0];
+
+  // assert userParams is an object and is not null
+  const emptyObject = {};
+  expect(userParams).not.toBeNull();
+  expect(userParams).toEqual(emptyObject);
+};
+
+/**
+ * Verifies that when a user does not provide the required parameters for a method, the validation step casues the method to fail.
+ * The `getMissingParams` mock function is expected to return an error and the code is not expected to ever call the `createRequest`
+ * method.
+ *
+ * @param {Object} missingParamsMock - the jest mock object for the `getMissingParams` function in  the `helpers` module
+ * @returns {void}
+ */
+module.exports.checkRequiredParamsHandling = function(
```

Comment:
This method is still confusing.  I was really only able to understand what it was doing by looking at the unit tests to see how they were using it.

Now that I do understand it, I think we probably want to change our unit tests so that we don't need this method.  I don't think we should mock `getMissingParameters` at all -- we can completely observe its behavior by inspecting the exception that is returned.  I'll elaborate this in the generator PR, but if we agree on that new approach, then we can simply remove this method from the core.